### PR TITLE
Add config migration for pr #145

### PR DESCRIPTION
**Pull request by oddstr13**

### CODE REVIEW

**Comment by oddstr13** [comment]
Well, I tested before pushing and it WorksForMe™©

**Comment by mcarlton00** [comment]
So using LocalAddress is a bad plan for those of us in docker.  It pulls the docker IP instead of the one I've manually defined. (which is `192.168.0.200:8097`)

```
2019-12-13 18:29:46.946 T:139730268165888  NOTICE: JELLYFIN.jellyfin.connection_manager -> INFO::jellyfin_kodi/jellyfin/connection_manager.py:161 Begin connect
2019-12-13 18:29:46.946 T:139730268165888  NOTICE: JELLYFIN.jellyfin.connection_manager -> INFO::jellyfin_kodi/jellyfin/connection_manager.py:69 Begin getAvailableServers
2019-12-13 18:29:47.952 T:139730268165888  NOTICE: JELLYFIN.jellyfin.connection_manager -> INFO::jellyfin_kodi/jellyfin/connection_manager.py:288 Found Servers: []
2019-12-13 18:29:47.958 T:139730268165888  NOTICE: JELLYFIN.jellyfin.connection_manager -> INFO::jellyfin_kodi/jellyfin/connection_manager.py:219 Begin connectToServers, with 1 servers
2019-12-13 18:29:47.959 T:139730268165888  NOTICE: JELLYFIN.jellyfin.connection_manager -> INFO::jellyfin_kodi/jellyfin/connection_manager.py:147 begin connectToServer
2019-12-13 18:29:47.959 T:139730268165888  NOTICE: JELLYFIN.jellyfin.connection_manager -> INFO::jellyfin_kodi/jellyfin/connection_manager.py:247 tryConnect url: http://172.17.0.2:8096/emby/system/info/public
2019-12-13 18:29:58.036 T:139730268165888  NOTICE: Jellyfin.jellyfin.http -> ERROR::jellyfin_kodi/jellyfin/http.py:102 HTTPConnectionPool(host='172.17.0.2', port=8096): Max retries exceeded with url: /emby/system/info/public (Caused by ConnectTimeoutError(<urllib3.connection.HTTPConnection object at 0x7f157d1a3690>, 'Connection to 172.17.0.2 timed out. (connect timeout=10)'))
2019-12-13 18:29:58.039 T:139730268165888  NOTICE: JELLYFIN.jellyfin.connection_manager -> ERROR::jellyfin_kodi/jellyfin/connection_manager.py:197 'auth.server-id'
                                            Traceback (most recent call last):
                                              File "jellyfin_kodi/jellyfin/connection_manager.py", line 195, in _request_url
                                                return self.http.request(request)
                                              File "jellyfin_kodi/jellyfin/http.py", line 103, in request
                                                self.client.callback("ServerUnreachable", {'ServerId': self.config.data['auth.server-id']})
                                            KeyError: 'auth.server-id'
2019-12-13 18:29:58.040 T:139730268165888  NOTICE: JELLYFIN.jellyfin.connection_manager -> INFO::jellyfin_kodi/jellyfin/connection_manager.py:157 Failing server connection. ERROR msg: 'auth.server-id'
2019-12-13 18:29:58.052 T:139730268165888  NOTICE: JELLYFIN.connect -> ERROR::jellyfin_kodi/connect.py:67 User is not authenticated.
```

File: jellyfin_kodi/database/__init__.py [hunk 1]
```diff
@@ -392,6 +392,15 @@ def get_credentials():
 
     credentials['Servers'] = credentials.get('Servers', [])
 
+    # Migration for #145
+    # TODO: CLEANUP for 1.0.0 release
+    for server in credentials['Servers']:
+        if 'LocalAddress' in server:
+            server['address'] = server['LocalAddress']
+            del server['LocalAddress']
```
Comment (mcarlton00) [inline jellyfin_kodi/database/__init__.py]: We should probably take a user defined address over the autodetected.  Maybe delete `LocalAddress` if `ManualAddress` exists too?
```
if 'ManualAddress' in server:
    server['address'] = server['ManualAddress']
    del server['ManualAddress']
elif 'LocalAddress' in server:
    server['address'] = server['LocalAddress']
    del server['LocalAddress']
```
I just tested and this worked for me

Comment (dkanada) [reply jellyfin_kodi/database/__init__.py]: If you're adding a check for LocalAddress might as well include RemoteAddress as a final fallback.

Comment (mcarlton00) [reply jellyfin_kodi/database/__init__.py]: Fair point.  We pulled the remote address out back in august with the wan IP in the server, but i'm sure some people have installs from that long that can still have it in the config.

**Comment by mcarlton00** [comment]
I added migrations for manual address and removed any references to the (very depreciated) remote address field.  Tested on a few of my older systems and all looks clean now

File: jellyfin_kodi/database/__init__.py [hunk 2]
```diff
@@ -392,6 +392,24 @@ def get_credentials():
...
+    for server in credentials['Servers']:
+        if 'ManualAddress' in server:
+            server['address'] = server['ManualAddress']
+            del server['ManualAddress']
+            # If manual is present, local should always be here, but better to be safe
+            if 'LocalAddress' in server:
+                del server['LocalAddress']
+            # Functionality removed in #60
+            if 'RemoteAddress' in server:
+                del server['RemoteAddress']
```
Comment (oddstr13) [inline jellyfin_kodi/database/__init__.py]: Not a big deal, but `RemoteAddress` does not get cleaned up if `ManualAddress` does not exist.

Ideally, this block would be at the same level as `LastConnectedMode`.

**Pull request merged**